code/game/objects/items/stacks/stack_recipes.dm

```diff
@@ -172,11 +172,30 @@
 	if(inherit_material)
 		var/datum/material/mat
 		var/datum/materials/materials_list = new
+
+		//Figure out the material
 		if(istype(S, /obj/item/stack/sheet/))
 			var/obj/item/stack/sheet/SS = S
 			mat = materials_list.getMaterial(SS.mat_type)
 		else if(S.material_type)
 			mat = S.material_type
+
+		// Make it recyclable back into the materials it's made out of
+		// Initialize materials list if doesn't exist already
+		if (R.materials == null)
+			R.materials = new /datum/materials(src)
+
+		// Add main materials off the stack
+		R.materials.addRatioFrom(S.materials, req_amount/(S.amount * res_amount))
+
+		// Add extra materials off additional recipe requisites
+		for (var/req in other_reqs)
+			// other_reqs contains typepaths, so create an instance and use it's materials as base
+			// TODO: pull the materials from the actual object that was used to fulfill the other_req
```

Comment:
Pointing this out right now: The materials added to an item by secondary requirements in a recipe, (eg: the plasteel in a tower shield) are decided by whatever materials a brand new instance of that requirement has. So, unrealistic example as it's not possible (yet), but if I were to craft a tower shield with plasteel that has been adulterated to contain uranium, said shield would fail to contain said uranium.
Hence the TODO, which I've no idea how to fix without digging into and possibly reworking how crafting recipes work in general.